cabal-install/Distribution/Client/Run.hs

```diff
@@ -42,23 +45,60 @@ import System.FilePath                       ((<.>), (</>))
 -- forwarded to it.
 splitRunArgs :: LocalBuildInfo -> [String] -> IO (Executable, [String])
 splitRunArgs lbi args =
-  case exes of
-    []    -> die "Couldn't find any executables."
-    [exe] -> case args of
-      []                        -> return (exe, [])
-      (x:xs) | x == exeName exe -> return (exe, xs)
-             | otherwise        -> return (exe, args)
-    _     -> case args of
-      []     -> die $ "This package contains multiple executables. "
-                ++ "You must pass the executable name as the first argument "
-                ++ "to 'cabal run'."
-      (x:xs) -> case find (\exe -> exeName exe == x) exes of
-        Nothing  -> die $ "No executable named '" ++ x ++ "'."
-        Just exe -> return (exe, xs)
+  case whichExecutable of -- Either err (wasManuallyChosen, exe, restParams)
+    Left err               -> do printMaybeWarningWithAdditional ""
+                                 die err
+    Right (True, exe, xs)  -> return (exe, xs)
+    Right (False, exe, xs) -> do printMaybeWarningWithAdditional
+                                   $ " Interpreting all parameters to `run`"
```

Comment:
I think you should use a trailing whitespace in `printMaybeWarningWithAdditional` instead of a prefix whitespace here (since this can be only message that gets printed).